### DISCUSSION TITLE
Questions on T-Deck, iOS, desktop (Mac), build my own? and more

### DISCUSSION THREAD

**Discussion by andypiper** [shorepine/tulipcc#142] [General]
I've been aware of the TulipCC project for many months, and with the T-Deck build becoming available, I've finally gotten to trying things out! As a MicroPython fan, it is exciting to have a pocketable device that runs it 🥳 

I've got a build installed on the T-Deck and things seem fine, apart from the keyboard frustrations noted in the README. On that note, have you looked at some of the other work going on around the T-Deck? At least one project has a [replacement sketch for the ESP32-C3 which appears to have added the Tab key](https://github.com/moononournation/T-Deck/blob/main/Keyboard_ESP32C3/Keyboard_ESP32C3.ino), for example (although, I've yet to play with this myself - the kit I received did contain the headers for the pinout at the edge of the keyboard which should provide access to reflash that, though).

What are the plans / what is the process for enabling access to additional hardware features on the T-Deck from this point onwards? I don't see a specific branch or anything, is the development happening directly on HEAD as things are stable?

For the iOS build, I'm struggling a little bit... although this may be due to having just updated to all the latest Apple OSes (e.g. I noticed that the simulator script looks for an iPhone 13 simulator, and I've got the iPhone 15 Pro now). For both the `build_simulator` and `build_device` scripts I'm hitting an error related to SDL not being found:

```text
GEN build-standard/tulip/obj/genhdr/qstr.i.last
main.c:565:10: fatal error: 'SDL_main.h' file not found
#include "SDL_main.h"
         ^~~~~~~~~~~~
1 error generated.
../shared_desktop/unix_display.c:2:10: fatal error: 'SDL.h' file not found
#include <SDL.h>
         ^~~~~~~
1 error generated.
../shared/keyscan.c:5:10: fatal error: 'SDL.h' file not found
#include <SDL.h>
         ^~~~~~~
1 error generated.
Traceback (most recent call last):
  File "/Users/andypiper/Development/third-party/tulipcc/tulip/ios/../../micropython/py/makeqstrdefs.py", line 211, in <module>
    preprocess()
...
...
```

I also had to manually download the SDL zip file - for some reason the build script wasn't hitting the clause that ran the curl and unpack - which got things further but evidently there's still a broken include path somewhere here. Have I missed a step or made a wrong assumption?

The Mac build seems to run fine here - I assume that `tulip.wifi()` fails due to MicroPython not providing access to the network interface in the Unix port?

Finally - if I wanted to go down the road of building my own "desktop" Tulip computer, has anyone had JLCPCB or similar do the complete assembly for them? I'm not super confident with surface mount soldering, but I have ordered full assemblies for other devices from JLCPCB before, and wondered whether anyone had been successful with the Tulip through there?

Sorry for asking a lot of questions all in one go and in one post - I wasn't sure if there's an active Discord or other place to ask things - I'm in a full rabbithole of investigation right now!

Great work on Tulip - it's a fun project!

**Comment by bwhitman** [comment]
> I've got a build installed on the T-Deck and things seem fine, apart from the keyboard frustrations noted in the README. On that note, have you looked at some of the other work going on around the T-Deck? At least one project has a [replacement sketch for the ESP32-C3 which appears to have added the Tab key](https://github.com/moononournation/T-Deck/blob/main/Keyboard_ESP32C3/Keyboard_ESP32C3.ino), for example (although, I've yet to play with this myself - the kit I received did contain the headers for the pinout at the edge of the keyboard which should provide access to reflash that, though).

Very cool! I'll play with it. My concern was if i asked users to both flash the T-Deck _and_ use a serial header to flash the keyboard, it may be too hard for anyone to really get going. 

> What are the plans / what is the process for enabling access to additional hardware features on the T-Deck from this point onwards? I don't see a specific branch or anything, is the development happening directly on HEAD as things are stable?

Dev (by me only right now) is on HEAD, and the next thing I'm doing is getting LoRA to work at the same time as the screen. I could use some help there, i think there's some conflict with MP's treatment of SPI vs ESP-iDFs. Link on the MP discord: https://discord.com/channels/574275045187125269/952412628926472273/1155667347688923286 

> For the iOS build, I'm struggling a little bit... although this may be due to having just updated to all the latest Apple OSes (e.g. I noticed that the simulator script looks for an iPhone 13 simulator, and I've got the iPhone 15 Pro now). For both the `build_simulator` and `build_device` scripts I'm hitting an error related to SDL not being found

This is mostly likely just an OS  update issue. I'll try playing with this over the next few days. 

> The Mac build seems to run fine here - I assume that `tulip.wifi()` fails due to MicroPython not providing access to the network interface in the Unix port?

You dont' need to run `tulip.wifi()` on Tulip Desktop on any desktop platform (iOS, macOS, Linux, Windows WSL) -- the network "just works."


> Finally - if I wanted to go down the road of building my own "desktop" Tulip computer, has anyone had JLCPCB or similar do the complete assembly for them? I'm not super confident with surface mount soldering, but I have ordered full assemblies for other devices from JLCPCB before, and wondered whether anyone had been successful with the Tulip through there?

There is a very smart man named @nvillar  who is doing this for us. I'll defer to him here or on Discord!

**Comment by nvillar** [comment]
Hi @andypiper , with respect to getting things made by JLCPCB - in theory everything is there. I haven't set up a JLCPCB assembly project yet, although hoping to do so soon. If you want to try it yourself in the meantime:

- The gerbers in `docs/pcbs/tulip4_board_v4r9/tulipcc-gerbers` are already in the correct format for JLCPB. I've made the test bare boards with them so I know they work. It's a two layer board, and the default JLCPCB settings worked (although I selected ENIG over HASL, although HASL should work just fine)

- The BOM has all the order numbers for Mouser, and also Digikey if they stock the part. JLCPCB can get these on consignment. It would be nicer and cheaper to user parts from the JLCPCB catalog, although last I checked some of the critical components (including the ESP32 S3 module) were out of stock in JLCPCB

**Reply by andypiper** [reply]
thanks for the feedback, I'll check it out when I can.
